jwst/resample/resample_utils.py

```diff
@@ -163,11 +163,18 @@ def reproject(wcs1, wcs2):
     """
 
     try:
+        # Here we want to use the WCS API functions so that a Sliced WCS
+        # will work as well. However, the API functions do not accept
+        # keyword arguments and `with_bounding_box=False` cannot be passsed.
+        # We delete the bounding box on a copy of the WCS - yes, inefficient.
         forward_transform = wcs1.pixel_to_world_values
-        backward_transform = wcs2.world_to_pixel_values
+        wcs_no_bbox = deepcopy(wcs2)
```

Comment:
I thought `deepcopy`ing a WCS is cpu-intense. Wouldn't it be better to you `try-finally` to turn it off/on? Still, I think @emolter 's comment about turning this off may be hiding some other issue is relevant.